data/src/test/java/org/apache/iceberg/RecordWrapperTest.java

```diff
@@ -103,11 +103,12 @@ public void testNestedSchema() {
   }
 
   private void generateAndValidate(Schema schema) {
-    generateAndValidate(schema, Assert::assertEquals);
+    generateAndValidate(
+        schema, (actual, message, expected) -> assertThat(actual).as(message).isEqualTo(expected));
   }
 
   public interface AssertMethod {
-    void assertEquals(String message, StructLikeWrapper expected, StructLikeWrapper actual);
+    void assertEquals(StructLikeWrapper actual, String message, StructLikeWrapper expected);
```

Comment:
it would be better to just keep this signature for now as otherwise it's causing lots of unrelated changes across Spark/Flink. That being said, focusing on the `data` module is the right thing here and seems to be the scope of your PR

Reply:
Got it, reverted this change.